core/src/Kokkos_Core_fwd.hpp

```diff
@@ -184,6 +186,15 @@ using DefaultHostExecutionSpace KOKKOS_IMPL_DEFAULT_HOST_EXEC_SPACE_ANNOTATION =
     "At least one of the following execution spaces must be defined in order to use Kokkos: Kokkos::OpenMP, Kokkos::Threads, Kokkos::Experimental::HPX, or Kokkos::Serial."
 #endif
 
+#if defined(KOKKOS_ENABLE_CUDA)
+using SharedSpace KOKKOS_IMPL_SHARED_SPACE_ANNOTATION = CudaUVMSpace;
+#elif defined(KOKKOS_ENABLE_HIP)
+using SharedSpace KOKKOS_IMPL_SHARED_SPACE_ANNOTATION = HIPManagedSpace;
+#elif defined(KOKKOS_ENABLE_SYCL)
+using SharedSpace KOKKOS_IMPL_SHARED_SPACE_ANNOTATION =
+    Experimental::SYCLSharedUSMSpace;
+#endif
```

Comment:
We had discussed the need for both a macro and a boolean variable that can be used in constant expressions to detect the existence of that alias

Reply:
I should have written this down :-D added in [c28db5](https://github.com/kokkos/kokkos/pull/5289/commits/c28db58c5cb9bc892226a542633e2eeeb440b2ee)

Reply:
I actually want in pure Host builds SharedSpace == HostSpace. Reasoning: it is shared and migrates to every existing ExecutionSpace and thus behaves performance wise basically as expected.

Reply:
done